src/lib/consensus/proof_of_stake.ml

```diff
@@ -2515,24 +2515,10 @@ module Hooks = struct
       |> Option.value
            ~default:
              ( "default case"
-             , "candidate virtual min-length is longer than existing virtual \
-                min-length"
+             , "candidate length is longer than existing length"
              , lazy
-                 (let newest_epoch =
-                    Epoch.max existing.curr_epoch candidate.curr_epoch
-                  in
-                  let virtual_min_length (s : Consensus_state.Value.t) =
-                    if Epoch.(succ s.curr_epoch < newest_epoch) then
-                      Length.zero (* There is a gap of an entire epoch *)
-                    else if Epoch.(succ s.curr_epoch = newest_epoch) then
-                      Length.(
-                        min s.min_length_of_epoch s.curr_epoch_data.length)
-                      (* Imagine the latest epoch was padded out with zeroes to reach the newest_epoch *)
-                    else s.min_length_of_epoch
-                  in
-                  Length.(
-                    virtual_min_length existing < virtual_min_length candidate))
-             )
+                 (* TODO: Undo this hack once the min_length_of_epoch bug is fixed *)
```

Comment:
Can you include the issue number? Also maybe add "This is insecure" or something to the effect in the comment.

Reply:
Issue is #2643